src/metatensor/models/experimental/alchemical_model/tests/test_exported.py

```diff
@@ -0,0 +1,33 @@
+import pytest
+import torch
+from metatensor.torch.atomistic import ModelCapabilities, ModelOutput
+
+from metatensor.models.experimental.alchemical_model import DEFAULT_HYPERS, Model
+from metatensor.models.utils.export import export
+from metatensor.models.utils.model_io import load_exported_model
+
+
+@pytest.mark.parametrize("device", ["cpu", "cuda"])
+@pytest.mark.parametrize("dtype", [torch.float32, torch.float64])
+def test_to(monkeypatch, tmp_path, device, dtype):
```

Comment:
```suggestion
def test_to(tmpdir device, dtype):
```

Why not just `tmpdir`?

Reply:
And then `with tmpdir.as_cwd():` https://github.com/lab-cosmo/metatensor/blob/1046d75ee0dfa4aaa4b6e00aface6bf89595063c/python/metatensor-learn/tests/dataset.py#L27

Reply:
Not sure. I'll change it

Reply:
It is not recommended to use `tmpdir` anymore. That is why we are using `monkeypatch` and `tmp_path`.

https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#the-tmpdir-and-tmpdir-factory-fixtures

Reply:
Because `tmp_path` can't be used as `with tmp_path.as_cwd():` because it has no context manager to change the directory. One could use os instead though.

Reply:
That kinda sucks for readability of the tests =/ Yeah, I'd go with `os.chdir(tmp_path)` then, we don't really need the `__exit__` part of the context manager